cmd/kubelet/app/server.go

```diff
@@ -289,11 +292,43 @@ func Run(s *options.KubeletServer, kcfg *KubeletConfig) error {
 }
 
 func run(s *options.KubeletServer, kcfg *KubeletConfig) (err error) {
+	if s.Bootstrap && s.LockFilePath == "" {
+		return errors.New("unable to start in bootstrap mode: no lock file provided")
+	}
+
+	done := make(chan struct{})
 	if s.LockFilePath != "" {
 		glog.Infof("aquiring lock on %q", s.LockFilePath)
 		if err := flock.Acquire(s.LockFilePath); err != nil {
 			return fmt.Errorf("unable to aquire file lock on %q: %v", s.LockFilePath, err)
 		}
+		if s.Bootstrap {
+			glog.Infof("watching for inotify events for: %v", s.LockFilePath)
+			watcher, err := inotify.NewWatcher()
+			if err != nil {
+				glog.Infof("unable to create watcher for lockfile: %v", err)
+				return err
+			}
+			if err := watcher.Watch(s.LockFilePath); err != nil {
+				glog.Infof("unable to watch lockfile: %v", err)
+				return err
+			}
+			go func() {
+				for {
+					select {
+					case ev := <-watcher.Event:
+						glog.Infof("inotify event: %v", ev)
+						if ev.Mask&inotify.IN_OPEN == inotify.IN_OPEN {
```

Comment:
Should be able to watch just for this event: `watcher.AddWatch(s.LockFilePath, inotify.IN_OPEN)`